src/LiquidityPool.sol

```diff
@@ -7,28 +7,26 @@ interface TrancheTokenLike {
     // erc20 functions
     function mint(address owner, uint256 amount) external;
     function burn(address owner, uint256 amount) external;
-
     function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
     function transfer(address recipient, uint256 amount) external returns (bool);
-    // function approve(address spender, uint256 amount) external returns (bool);
     function approveForOwner(address owner, address spender, uint256 value) external returns (bool);
-
     function totalSupply() external view returns (uint256);
     function balanceOf(address owner) external returns (uint256);
     function allowance(address owner, address spender) external returns (uint256);
-    // function increaseAllowance(address spender, uint256 addedValue) external returns (bool);
-    // function decreaseAllowance(address spender, uint256 subtractedValue) external returns (bool);
     function increaseAllowanceForOwner(address owner, address spender, uint256 addedValue) external returns (bool);
     function decreaseAllowanceForOwner(address owner, address spender, uint256 subtractedValue)
         external
         returns (bool);
-
     // restricted token functions
     function latestPrice() external view returns (uint256);
     function memberlist() external returns (address);
     function hasMember(address) external returns (bool);
 }
 
+interface ERC20Like {
+    function allowance(address owner, address spender) external returns (uint256);
+}
+
 interface InvestmentManagerLike {
```

Comment:
Another unrelated comment: I think we should consider making `share` and `asset` immutable. If these somehow need to be changed, I think it would make anyways more sense to deploy new liquidity pools rather than actually fiel them.